### PR TITLE
Implement W3C Protocol fallback for Click Commands / Button Protocols

### PR DESCRIPTION
**Pull request by emilyrohrbough**
## Proposed changes

This change adds W3C fallback support for drivers that have implemented the actions API and have dropped JSONWireframe support for protocol endpoints `buttonDown`, `buttonPress` and `buttonUp`.  This directly affects the `leftClick`, `middleClick` and `rightClick` commands.

New drivers are at different implementation stages of the W3C protocol implementation. IE and Firefox both support the actions API, however, Chrome does not yet support actions API.

Addresses https://github.com/webdriverio/webdriverio/issues/3107

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to WebdriverIO?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [ ] I have proposed the same patch to the new [v5](https://github.com/webdriverio/v5) repository
     - N/A: the `leftClick`, `middleClick` and `rightClick` commands are currently removed from v5. [Indicated in v5 Changelog](https://github.com/webdriverio/webdriverio/blob/1dfe9442a8fdcf5bf0a279b965047cd98eb35e8b/CHANGELOG.md#v500-2018-12-20)

## Further comments

[//]: # (If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...)

### CODE REVIEW

**Comment by emilyrohrbough** [comment]
Any feedback on this?

**Comment by emilyrohrbough** [comment]
I hate to be a pain, but this change is blocking us. Can I get this merged?

**Comment by ryanthemanuel** [comment]
+1 to this.  My team is needing this as well.

**Comment by bjankord** [comment]
This addition would be helpful for my projects as well. Can we get this reviewed / merged @christian-bromann

**Comment by benbcai** [comment]
+1 This is a blocker for me as well.

**Comment by emilyrohrbough** [comment]
@christian-bromann @abjerstedt @pmerwin Any feedback on this?

**Comment by emilyrohrbough** [comment]
Can I get some feedback on this PR? @christian-bromann @abjerstedt @pmerwin This is a blocker and would like to get some traction on this.

File: lib/helpers/handleMouseButtonProtocol.js [hunk 1]
```diff
@@ -15,7 +23,32 @@ let handleMouseButtonProtocol = function (requestPath, button) {
     return this.requestHandler.create(
         requestPath,
         { button: button }
-    )
+    ).catch(err => {
+        /**
+         * jsonwire command not supported try webdriver endpoint
+         */
+        if (ProtocolError(err)) {
```
Comment (christian-bromann) [inline lib/helpers/handleMouseButtonProtocol.js]: This will be always true, no?

**Review by christian-bromann — CHANGES_REQUESTED** [review]
I think the if statement as it is right now doesn't work

File: lib/helpers/handleMouseButtonProtocol.js (hunk 1 above)
Comment (pmerwin) [reply lib/helpers/handleMouseButtonProtocol.js]: Well, isn't this only true of a "ProtoclError", so if there is another error type we won't enter in here?  Is that why?

Comment (emilyrohrbough) [reply lib/helpers/handleMouseButtonProtocol.js]: Good catch, this should actually be checking for unknown command.

**Comment by emilyrohrbough** [comment]
I created an [isolated test spec](https://github.com/cerner/terra-toolkit/blob/reduced-test-case/tests/wdio/button-press-spec.js) for those interested to test [this page](https://github.com/cerner/terra-toolkit/blob/reduced-test-case/tests/fixtures/test.html). This setups up a [selenium grid for chrome and firefox](https://github.com/cerner/terra-toolkit/blob/reduced-test-case/config/docker/local-docker-compose.yml). If desired you can pull this branch, run `npm install && npm run wdio` to hit the master branch vs this branch to verify these changes. 

When using my forked [master](https://github.com/emilyrohrbough/v4/tree/master) branch for the webdriver.io dependency, it will produce these errors:
screen shot 2019-02-20 at 2 31 05 pm


When updated to reference this forked branch [verify-w3c-button-fallback](https://github.com/emilyrohrbough/v4/tree/verify-w3c-button-fallback), these tests will pass:
screen shot 2019-02-20 at 2 56 24 pm

The reason these reference my forked branches is to run `npm run build` in webdriver.io's prepare script such that I can use wdio in my repo.

File: lib/helpers/handleMouseButtonProtocol.js (hunk 1 above)
Comment (emilyrohrbough) [reply lib/helpers/handleMouseButtonProtocol.js]: @christian-bromann @pmerwin  Do these changes look good?

**Comment by emilyrohrbough** [comment]
@christian-bromann @pmerwin Any other feedback on this PR?

**Comment by bjankord** [comment]
This would help unblock our team! Thanks for contributing this PR @emilyrohrbough!

**Comment by benbcai** [comment]
When can this be reviewed and merged? I'm still in need of this change.

**Review by christian-bromann — APPROVED** [review]
👍

**Comment by emilyrohrbough** [comment]
Thank you @christian-bromann !!

**Pull request merged**